File: CS/LinqServerModeOnAdvWorks/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;

namespace LinqServerModeOnAdvWorks {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            gridControl1.ServerMode = true;

            AdvWorksDataContext dc = new AdvWorksDataContext();
            dc.Log = Console.Out;  // log queries

            linqServerModeSource1.QueryableSource = dc.PurchaseOrderHeaders;
            gridControl1.DataSource = linqServerModeSource1;
        }
    }
}

File: VB/LinqServerModeOnAdvWorks/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace LinqServerModeOnAdvWorks

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            gridControl1.ServerMode = True
            Dim dc As AdvWorksDataContext = New AdvWorksDataContext()
            dc.Log = Console.Out ' log queries
            linqServerModeSource1.QueryableSource = dc.PurchaseOrderHeaders
            gridControl1.DataSource = linqServerModeSource1
        End Sub
    End Class
End Namespace
